### PR TITLE
Enable "-race" option for telemetry repo testing.

### PR DESCRIPTION
**Pull request by FengPan-Frank**
#### Why I did it
Enable golang race detector in telemetry repo.

#### How I did it
Always run outer loop tests with "-race" enabled

#### How to verify it
Use unit test.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -62,21 +62,21 @@ go-deps-clean:
...
-	$(GO) install -mod=vendor github.com/openconfig/gnmi/cmd/gnmi_cli
-	$(GO) install -mod=vendor github.com/sonic-net/sonic-gnmi/gnoi_client
-	$(GO) install -mod=vendor github.com/sonic-net/sonic-gnmi/gnmi_dump
+	$(GO) install -race -mod=vendor $(BLD_FLAGS) github.com/sonic-net/sonic-gnmi/telemetry
+	$(GO) install -race -mod=vendor $(BLD_FLAGS) github.com/sonic-net/sonic-gnmi/dialout/dialout_client_cli
+	$(GO) install -race -mod=vendor github.com/jipanyang/gnxi/gnmi_get
+	$(GO) install -race -mod=vendor github.com/jipanyang/gnxi/gnmi_set
+	$(GO) install -race -mod=vendor github.com/openconfig/gnmi/cmd/gnmi_cli
+	$(GO) install -race -mod=vendor github.com/sonic-net/sonic-gnmi/gnoi_client
+	$(GO) install -race -mod=vendor github.com/sonic-net/sonic-gnmi/gnmi_dump
```
Comment (ganglyu) [inline Makefile:79]: Please check runtime overheads:
https://go.dev/doc/articles/race_detector#Runtime_Overheads
Do we really need to use race detector for go build and go install?

**Comment by prsunny** [comment]
@theasianpianist for viz

File: Makefile (hunk 1 above)
Comment (FengPan-Frank) [reply Makefile:79]: Right, as formal change we may need to use separate telemetry binary without "-race" enable into product build

**Comment by FengPan-Frank** [comment]
Use https://github.com/sonic-net/sonic-gnmi/pull/100 instead.

**Pull request closed without merging**